### DISCUSSION TITLE
What does the "scheduler thread" in the architecture doc refer to ?

### DISCUSSION THREAD

**Discussion by RedemptionC** [apache/dolphinscheduler#8279] [Q&A]
As far as I know, there are 2 kind of scheduler in DS:
1. the one that's related to quartz
2. the MasterSchedulerService

I read both English and Chinese version of the doc in 1.3.9 version,and they use the term like this:
[Chinese](https://github.com/apache/dolphinscheduler-website/blob/master/docs/zh-cn/1.3.9/user_doc/architecture-design.md#%E5%9B%9B%E5%AE%B9%E9%94%99%E8%AE%BE%E8%AE%A1):
>ZooKeeper Master容错完成之后则重新由DolphinScheduler中Scheduler线程调度，遍历 DAG 找到”正在运行”和“提交成功”的任务

[English](https://github.com/apache/dolphinscheduler-website/blob/master/docs/en-us/1.3.9/user_doc/architecture-design.md#fourfault-tolerant-design):
>After the fault tolerance of ZooKeeper Master is completed, it is re-scheduled by the Scheduler thread in DolphinScheduler

As I see,it's more likely to refer to the second one: the MasterSchedulerService

But I have read the code of MasterSchedulerService,it just retrive a command from DB,and execute that 
I'm confused

what's your opinion? @SunnyZ-L  @zhongjiajie

**Comment by zhongjiajie** [comment]
I'm not familiar with master server, could you please take a look @caishunfeng

**Comment by caishunfeng** [comment]
`Command` as the unified entrance and `MasterSchedulerService` query command and convert to processInstance.
You can find the main control logic of workflow in `WorkflowExecuteThread`.
[architecture design doc](https://dolphinscheduler.apache.org/zh-cn/docs/latest/user_doc/architecture/design.html)
